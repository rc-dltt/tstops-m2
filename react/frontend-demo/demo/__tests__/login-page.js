import React from "react";
import { MockedProvider } from '@apollo/client/testing';
import { PaperProvider } from 'react-native-paper';
import LoginPage from "../Components/LoginPage";
import { loginMutation } from '../graphql/mutation';
import { fireEvent, cleanup, render, screen, renderHook } from "@testing-library/react-native";
import { act } from "react-test-renderer";

const mocks = [
    {
        request: {
            query: loginMutation,
            variables: {
                email: 'johndoe@email.com',
                password: 'pAsSWoRd!',
            },
        },
        result: {
            data: {
                login: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6InVzZXIjMSIsInJvbGVzIjpbInVzZXIiLCJhZG1pbiJdLCJpYXQiOjE2OTI2NzE5NTEsImV4cCI6MTY5Mjc1ODM1MSwic3ViIjoidXNlciMxIn0.3CPT3fQqU6fLarIh8y0zFkJD7Xik5UdzBBkapqZdzSM"
            },
        },
    },
];

beforeEach(() => {

    jest.useFakeTimers();
});

describe('Login Page', () => {
    it('Snapshot', () => {
        const handleLogin = jest.fn();

        const loginPage = render(
            <MockedProvider mocks={mocks} addTypename={false} >
                <PaperProvider>
                    <LoginPage onLogin={handleLogin} />
                </PaperProvider>
            </MockedProvider>
        );
        expect(loginPage).toMatchSnapshot();
    })

    it('Login Input Change', () => {

        const handleLogin = jest.fn();

        const { getByTestId, getByDisplayValue } = render(
            <MockedProvider mocks={mocks} addTypename={false} >
                <PaperProvider>
                    <LoginPage onLogin={handleLogin} />
                </PaperProvider>
            </MockedProvider>
        );

        const email = 'johndoe@email.com';
        const pw = 'pAsSWoRd!';

        const emailInput = getByTestId('email-input');
        const passwordInput = getByTestId('password-input');
        fireEvent.changeText(emailInput, email);
        fireEvent.changeText(passwordInput, pw);

        const emailInputChanged = getByDisplayValue(email);
        const pwInputChanged = getByDisplayValue(pw);

        expect(emailInputChanged).toBeTruthy();
        expect(pwInputChanged).toBeTruthy();
    })

    it('Login Submit', async () => {

        const handleLogin = jest.fn();

        const { getByTestId, getByDisplayValue } = render(
            <MockedProvider mocks={mocks} addTypename={false} >
                <PaperProvider>
                    <LoginPage onLogin={handleLogin} />
                </PaperProvider>
            </MockedProvider>
        );

        const email = 'johndoe@email.com';
        const pw = 'pAsSWoRd!';
        const emailInput = getByTestId('email-input');
        const passwordInput = getByTestId('password-input');
        const loginButton = getByTestId('login-btn');
        fireEvent.changeText(emailInput, email);
        fireEvent.changeText(passwordInput, pw);
        const emailInputChanged = getByDisplayValue(email);
        const pwInputChanged = getByDisplayValue(pw);
        expect(emailInputChanged).toBeTruthy();
        expect(pwInputChanged).toBeTruthy();
        act(()=> {
            fireEvent.press(loginButton);
        })

    })
})
