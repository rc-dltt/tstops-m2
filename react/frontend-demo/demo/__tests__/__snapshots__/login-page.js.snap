// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTSafeAreaView
      style={
        {
          "backgroundColor": "#F3F3F3",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../img/hkjc-icon.png",
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(247, 243, 249, 1)",
            "borderRadius": 12,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.15,
            "shadowRadius": 3,
          }
        }
        testID="card-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(247, 243, 249, 1)",
              "borderRadius": 12,
              "elevation": 1,
              "flex": undefined,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            }
          }
          testID="card-container"
        >
          <View
            style={
              [
                {
                  "flexShrink": 1,
                },
                undefined,
              ]
            }
            testID="card"
          >
            <View
              borderRadiusStyles={{}}
              style={
                [
                  {
                    "paddingHorizontal": 16,
                  },
                  {
                    "paddingVertical": 16,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#424d9c",
                    "fontSize": 24,
                    "fontWeight": "bold",
                    "margin": 12,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                  }
                }
                variant="titleLarge"
              >
                Login
              </Text>
              <View
                style={
                  [
                    {
                      "backgroundColor": "rgba(231, 224, 236, 1)",
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    {
                      "margin": 12,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "rgba(28, 27, 31, 1)",
                      "bottom": 0,
                      "height": 1,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "transform": [
                        {
                          "scaleY": 0.5,
                        },
                      ],
                      "zIndex": 1,
                    }
                  }
                  testID="text-input-underline"
                />
                <View
                  style={
                    [
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "minHeight": 56,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    pointerEvents="none"
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                        ],
                        "zIndex": 3,
                      }
                    }
                  >
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      onLayout={[Function]}
                      style={
                        {
                          "color": "#424d9c",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 0,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 30,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                            {
                              "translateY": 0,
                            },
                            {
                              "scale": 1,
                            },
                          ],
                          "writingDirection": "ltr",
                        }
                      }
                      testID="email-input-label-active"
                    >
                      Email
                    </Text>
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      style={
                        {
                          "color": "rgba(73, 69, 79, 1)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 0,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 30,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                            {
                              "translateY": 0,
                            },
                            {
                              "scale": 1,
                            },
                          ],
                          "writingDirection": "ltr",
                        }
                      }
                      testID="email-input-label-inactive"
                    >
                      Email
                    </Text>
                  </View>
                  <TextInput
                    backgroundColor="#Eceffb"
                    cursorColor="#424d9c"
                    editable={true}
                    maxFontSizeMultiplier={1.5}
                    multiline={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder=" "
                    placeholderTextColor="rgba(73, 69, 79, 1)"
                    selectionColor="#424d9c"
                    style={
                      [
                        {
                          "margin": 0,
                        },
                        {
                          "paddingLeft": 16,
                          "paddingRight": 16,
                        },
                        {
                          "height": 56,
                        },
                        {
                          "paddingBottom": 4,
                          "paddingTop": 24,
                        },
                        {
                          "color": "#505f93",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        false,
                        [
                          {},
                        ],
                        undefined,
                      ]
                    }
                    testID="email-input"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                onLayout={[Function]}
                style={
                  {
                    "color": "rgba(179, 38, 30, 1)",
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "400",
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                    "opacity": 0,
                    "paddingHorizontal": 12,
                    "paddingVertical": 4,
                    "textAlign": "left",
                    "transform": [],
                    "writingDirection": "ltr",
                  }
                }
              />
              <View
                style={
                  [
                    {
                      "backgroundColor": "rgba(231, 224, 236, 1)",
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    {
                      "margin": 12,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "rgba(28, 27, 31, 1)",
                      "bottom": 0,
                      "height": 1,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "transform": [
                        {
                          "scaleY": 0.5,
                        },
                      ],
                      "zIndex": 1,
                    }
                  }
                  testID="text-input-underline"
                />
                <View
                  style={
                    [
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "minHeight": 56,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    pointerEvents="none"
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                        ],
                        "zIndex": 3,
                      }
                    }
                  >
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      onLayout={[Function]}
                      style={
                        {
                          "color": "#424d9c",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 0,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 30,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                            {
                              "translateY": 0,
                            },
                            {
                              "scale": 1,
                            },
                          ],
                          "writingDirection": "ltr",
                        }
                      }
                      testID="password-input-label-active"
                    >
                      Password
                    </Text>
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      style={
                        {
                          "color": "rgba(73, 69, 79, 1)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 0,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 30,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                            {
                              "translateY": 0,
                            },
                            {
                              "scale": 1,
                            },
                          ],
                          "writingDirection": "ltr",
                        }
                      }
                      testID="password-input-label-inactive"
                    >
                      Password
                    </Text>
                  </View>
                  <TextInput
                    backgroundColor="#Eceffb"
                    cursorColor="#424d9c"
                    editable={true}
                    maxFontSizeMultiplier={1.5}
                    multiline={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder=" "
                    placeholderTextColor="rgba(73, 69, 79, 1)"
                    secureTextEntry={true}
                    selectionColor="#424d9c"
                    style={
                      [
                        {
                          "margin": 0,
                        },
                        {
                          "paddingLeft": 16,
                          "paddingRight": 16,
                        },
                        {
                          "height": 56,
                        },
                        {
                          "paddingBottom": 4,
                          "paddingTop": 24,
                        },
                        {
                          "color": "#505f93",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        false,
                        [
                          {},
                        ],
                        undefined,
                      ]
                    }
                    testID="password-input"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                onLayout={[Function]}
                style={
                  {
                    "color": "rgba(179, 38, 30, 1)",
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "400",
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                    "opacity": 0,
                    "paddingHorizontal": 12,
                    "paddingVertical": 4,
                    "textAlign": "left",
                    "transform": [],
                    "writingDirection": "ltr",
                  }
                }
              />
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "#424d9c",
                    "borderRadius": 20,
                    "margin": 12,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
                testID="login-btn-container-outer-layer"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "#424d9c",
                      "borderColor": "transparent",
                      "borderRadius": 20,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "flex": undefined,
                      "minWidth": 64,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="login-btn-container"
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        {
                          "borderRadius": 20,
                        },
                      ]
                    }
                    testID="login-btn"
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          [
                            {
                              "textAlign": "left",
                            },
                            {
                              "color": "rgba(28, 27, 31, 1)",
                              "writingDirection": "ltr",
                            },
                            [
                              {
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              [
                                {
                                  "marginHorizontal": 16,
                                  "marginVertical": 9,
                                  "textAlign": "center",
                                },
                                false,
                                {
                                  "marginHorizontal": 24,
                                  "marginVertical": 10,
                                },
                                undefined,
                                false,
                                {
                                  "color": "#fbde2b",
                                  "fontFamily": "System",
                                  "fontSize": 14,
                                  "fontWeight": "500",
                                  "letterSpacing": 0.1,
                                  "lineHeight": 20,
                                },
                                undefined,
                              ],
                            ],
                          ]
                        }
                        testID="login-btn-text"
                      >
                        Login
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</RNCSafeAreaProvider>
`;
